packages/iris-grid/src/IrisGridPartitionSelector.tsx

```diff
@@ -1,316 +1,394 @@
 
     this.debounceUpdate.cancel();
 
-    const { onFetchAll } = this.props;
-    onFetchAll();
+    const { onMerge } = this.props;
+    onMerge();
   }
 
-  getDisplayValue(column: Column, index: number): string {
-    const { partitions } = this.state;
-    const partition = partitions[index];
-    if (partition == null) {
+  sendKeyTable(): void {
+    log.debug2('sendOpenKeys');
+
+    this.debounceUpdate.cancel();
+
+    const { onKeyTable } = this.props;
+    onKeyTable();
+  }
+
+  /** Calls getFormattedString with a special character case */
+  getDisplayValue(index: number, partition?: unknown): string {
+    const { columns, getFormattedString } = this.props;
+    const { partitionConfig } = this.state;
+
+    const value =
+      partition === undefined ? partitionConfig.partitions[index] : partition;
+    if (value == null || value === '') {
       return '';
     }
-    if (TableUtils.isCharType(column.type) && partition.toString().length > 0) {
-      return String.fromCharCode(parseInt(partition, 10));
+    const column = columns[index];
+    if (TableUtils.isCharType(column.type) && value.toString().length > 0) {
+      return String.fromCharCode(parseInt(value.toString(), 10));
     }
-    return IrisGridUtils.convertValueToText(partition, column.type);
+    return getFormattedString(value, column.type, column.name);
   }
 
-  async updatePartitionFilters(
-    partitions: (string | null)[],
-    partitionTables: Table[]
+  /**
+   * Resolve invalid partitions and get new values for partition dropdowns
+   * @param index The index of the partition that was changed
+   * @param partitions Array of partitions containing updated values
+   */
+  async updateConfig(
+    partitionConfig: PartitionConfig,
+    index = 0,
+    updateIrisGrid = false
   ): Promise<void> {
-    const { columns, getFormattedString } = this.props;
+    if (partitionConfig.mode !== 'partition') {
+      this.clearDropdowns(partitionConfig);
+      return;
+    }
+    if (!Array.isArray(this.partitionTables)) {
+      return;
+    }
+    log.debug('partitionSelector update', index, partitionConfig);
+    const { getFormattedString } = this.props;
+    // Cannot use columns from props since different index number will cause filters to fail
+    const { columns } = this.partitionTables[this.partitionTables.length - 1];
+    if (!this.table) {
+      // this.table should be assigned in componentDidMount before updatePartitions is called
+      throw new Error('Table not initialized');
+    }
+
+    const partitionFilters = [...this.partitionTables[index].filter];
+    const validPartitions = [...partitionConfig.partitions];
+    let lastValidPartition = null;
 
-    const partitionFilters = [];
-    for (let i = 0; i < columns.length - 1; i += 1) {
-      const partition = partitions[i];
+    // Update partition filters
+    for (let i = index; i < partitionConfig.partitions.length; i += 1) {
+      /* eslint-disable  no-await-in-loop */
```

Comment:
Add a comment about how you need to do the await in a loop because each subsequent partition list is based on the results from the previous one (cascading).
We also need to check that user didn't do something else that invalidates this work. Easiest way to do that is to set a loading state, disable the controls while in that state (since it's loading), await on updateConfig when you call it, and then unset the loading state when `updateConfig` has finished.